### PR TITLE
fix(Core/Scripts): Venomhide Hatchling improvements

### PR DESCRIPTION
**Pull request by avarishd**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

PR also corrects the ability to use other people's hatchling to do your quest/projectile to them.

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8409
- Closes https://github.com/chromiecraft/chromiecraft/issues/2473
- Closes https://github.com/chromiecraft/chromiecraft/issues/2488
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8107
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/9296
- Closes https://github.com/chromiecraft/chromiecraft/issues/2376
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/9548
- Closes https://github.com/chromiecraft/chromiecraft/issues/2497
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8779
- Closes https://github.com/chromiecraft/chromiecraft/issues/2610

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

```sql
-- modelid source
SELECT Name,modelid1 FROM `creature_template` WHERE `name` LIKE '%venomhide hatch%';
```
## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

growgif

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

-- Model Size
.q add 13906
.additem 47196 X (amount)

-- Using items
.additem 46380 / 46382 / 46367
Try to use items while summoned and when not, should only work while he is present.
And should have proper projectile,

-- Despawn on quest complete
.go c 88163
.q a 13906
.q c 13906
Summon the hatchling and turn in the quest, npc should despawn.

-- See if quest mark is shown when not on quest
.q add 13906
Jump on 2nd character without the quest, there shouldn't be an exclamation mark.

## Known Issues and TODO List:

- [ ] You can accept  [They Grow Up So Fast] (13906), summon the hatchling and abandon it and it will still stay, one solution would be quest check on a timer in the creature script, but that would be very, very wasteful.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by avarishd** [comment]
Need help figuring out why SpellVisual correction does nothing.

File: src/server/scripts/Spells/spell_item.cpp [hunk 1]
```diff
@@ -3883,6 +3883,64 @@ class spell_item_worn_troll_dice : public SpellScript
...
+        if (Player* player = GetCaster()->ToPlayer())
+        {
+            std::list<Creature*> hatchling;
+            player->GetAllMinionsByEntry(hatchling, NPC_VENOMHIDE_HATCHLING);
+            if (hatchling.empty())
+            {
+                return;
+            }
+
+            for (Creature* creature  : hatchling)
```
Comment (Kitzunu) [inline src/server/scripts/Spells/spell_item.cpp:3926]:
```suggestion
            for (Creature* creature : hatchling)
```

Comment (avarishd) [reply src/server/scripts/Spells/spell_item.cpp:3926]: I see nothing wrong.

Comment (Kitzunu) [reply src/server/scripts/Spells/spell_item.cpp]: literally a double space, as you can see in my suggested change

Comment (avarishd) [reply src/server/scripts/Spells/spell_item.cpp]: It was a joke

Comment (Kitzunu) [reply src/server/scripts/Spells/spell_item.cpp]: I am a joke 👀

**Review by Kitzunu — APPROVED** [review]

File: src/server/scripts/Spells/spell_item.cpp (hunk 1 above)
Comment (avarishd) [reply src/server/scripts/Spells/spell_item.cpp]: 🤔🤔🤔

**Pull request merged**